### ISSUE TITLE
MySQL received close signal, exiting  (aborted connection, got an error reading communication packets)

### ISSUE THREAD

**Issue by marinakr** [processone/ejabberd#2031]
> What version of ejabberd are you using?

16.06

> What operating system (version) are you using?

Ubuntu

> How did you install ejabberd (source, package, distribution)?

source

> What did not work as expected? Are there error messages in the log? What
> was the unexpected behavior? What was the expected result?

I have tried to use MySQL database. With postgres it works fine. But I can not connect to mysql databes
Config file:
```yaml
odbc_type: mysql
odbc_server: "localhost"
odbc_database: "ejabberd"
odbc_username: "ejabberd"
odbc_password: "ejabberd"
#odbc_pool_size: 10
## If you want to specify the port:
odbc_port: 3306
```

I can successfully connect to mysql database with mysql client from console:
`   mysql -h localhost -u ejabberd -p -D ejabberd`
```
Your MySQL connection id is 56
Server version: 5.7.19 MySQL Community Server (GPL)
```

From mysql error.log:
```
2017-10-03T16:09:34.455034Z 41 [Note] Aborted connection 41 to db: 'ejabberd' user: 'ejabberd' host: 'localhost' (Got an error reading communication packets)
2017-10-03T16:09:34.455191Z 44 [Note] Aborted connection 44 to db: 'ejabberd' user: 'ejabberd' host: 'localhost' (Got an error reading communication packets)
2017-10-03T16:09:34.455319Z 40 [Note] Aborted connection 40 to db: 'ejabberd' user: 'ejabberd' host: 'localhost' (Got an error reading communication packets)
```
Ejabberd log:
```
19:22:37.606 [error] p1_mysql_conn: Received close signal, exiting.
19:22:37.606 [error] p1_mysql_conn: Received close signal, exiting.
19:22:37.606 [info] Closing connection #Port<0.15569>: ok
19:22:37.606 [info] Closing connection #Port<0.15565>: ok
19:22:37.606 [error] p1_mysql_conn: Received close signal, exiting.
19:22:37.607 [info] Closing connection #Port<0.15570>: ok
```

**Comment by badlop** [comment]
Maybe the mysql library included in ejabberd 16.06 is not compatible with your MySQL server? Try with a recent ejabberd version, you can try a binary installer, just to check if that's the problem.

**Comment by marinakr** [comment]
Ok, I will check it in few days. MySQL server version is 5.7, should be compatible
